extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```
@@ -208,6 +216,64 @@ public KubernetesPortBuildItem kubernetesForManagement(
                 managementBuildTimeConfig.enabled());
     }
 
+    @BuildStep
+    void registerLocalStaticResources(VertxHttpBuildTimeConfig httpBuildTimeConfig,
+            BuildProducer<GeneratedStaticResourceBuildItem> generatedStaticResources) {
+
+        var localStatic = httpBuildTimeConfig.localStaticResources();
+        if (!localStatic.enabled()) {
+            return;
+        }
+
+        final String basePath = localStatic.normalizedPath();
+
+        Path root = Path.of(localStatic.directory()).normalize();
```

I believe we should also check this for `..`, FYI you can use `import jakarta.validation.constraints.Pattern;` directly in the config:
```java
    @Pattern(regexp = "^(?!.*\\.\\.).*$")
    @WithDefault("static")
    String directory();
```

---

Good idea, thanks!
Right now we only validate the URL path (via normalizedPath()), while directory is just passed through Path.of(...).normalize(), so .. is still allowed in the config.
I’ll add a @Pattern on directory() to forbid .. as you suggested, so we fail fast at configuration time.